src/clj/rems/form_validation.clj

```diff
@@ -52,6 +52,13 @@
       {:field-id (:field/id field)
        :type     :t.form.validation/invalid-phone-number})))
 
+(defn- invalid-ip-address-error [field]
+  (when (= (:field/type field) :ip-address)
+    (when-not (or (str/blank? (:field/value field))
+                  (re-matches +ip-address-regex+ (:field/value field)))
```

Comment:
what about optionally rejecting private IP addresses?

Reply:
can I have some sort of summary of what addresses are acceptable and what are not? will making testing much easier. @jaakkocsc ?

Reply:
The summary is actually in the ticket.

Reply:
I don't think the spec for the private ranges is in the ticket, but you can find it for example here:
https://en.wikipedia.org/wiki/Private_network#Private_IPv4_addresses

Reply:
Yes, that task is about investigating first and then doing a reasonable set.

Reply:
@Macroz 
```
 IP address should support ipv4 and ipv6
 Optionally reject private IP addresses (10.x.x.x etc.)
```

is the only thing I see in the ticket

Reply:
Yes, to me that is a summary of what needs to be done. I.e. support IPv4 and v6 (this so far has only IPv4), and investigate rejecting private IP address ranges, the details will be revealed by an investigation and we do not have the information yet. 

So the task includes the investigation. If we had exact information then it'd be in the ticket too. So this is the level of summary we can have without actually doing the investigation first. Perhaps you are asking for someone else to do the investigation?

Reply:
In addition to the "private" ranges, it's probably good to also exclude "link-local" ranges. https://en.wikipedia.org/wiki/Private_network#Link-local_addresses

Reply:
I ll create regexes and unit tests accordingly